pycqed/measurement/sweep_points.py

```diff
@@ -270,44 +270,59 @@ def get_meas_obj_sweep_points_map(self, measurement_objects):
-            the order of params in each sweep dimension corresponds to the
-            order of keys in keys_list.
-            I.e. key_i in sweep_points[0] contains the sweep information for
-            measured_objects[i].
-
-        :param measured_objects: list of strings to be used as keys in the
-            returned dictionary. These are the measured object names
+            - if some parameters remain that do not contain any of the
+            measurement_objects names, it is assumed that all
+            measurement_objects used them so they will be added for each mobj
+
+        :param measurement_objects: list of strings to be used as keys in the
+            returned dictionary. These are the measured object names.
+            Can also be list of QuDev_transmon instanced in which case
+            this function gets the list of names.
         :return: dict of the form
          {mobj_name: [sweep_param_name_0, ..., sweep_param_name_n]}
         """
+
+        # Ensure measurement_objects is a list
         if isinstance(measurement_objects, list):
             measurement_objects = copy(measurement_objects)
         else:
             measurement_objects = [measurement_objects]
+
         for i, mobj in enumerate(measurement_objects):
             if hasattr(mobj, 'name'):
+                # list of QuDev_transmon instances was provided
                 measurement_objects[i] = mobj.name
 
-        sweep_points_map = OrderedDict()
-        for i, mobjn in enumerate(measurement_objects):
-            sweep_points_map[mobjn] = []
-            for dim, d in enumerate(self):
-                if len(d) == 1:
-                    # assume all mobjs use the same param_name
+        sweep_points_map = {mobjn: [] for mobjn in measurement_objects}
+
+        for dim, d in enumerate(self):
+            # d is a dictionary with keys == sweep parameter names
+            if len(d) == 1:
+                # Only one sweep parameter.
+                # Assume all mobjs use the same param_name given by the key
+                # of d.
+                for mobjn in measurement_objects:
                     sweep_points_map[mobjn] += [next(iter(d))]
-                elif mobjn in list(d)[i]:
-                    sweep_points_map[mobjn] += [list(d)[i]]
-                else:
-                    if len(d) != len(measurement_objects):
-                        raise ValueError(
-                            f'{len(measurement_objects)} measurement objects '
-                            f'were given but there are {len(d)} '
-                            f'sweep parameters in dimension {dim}.')
-                    sweep_points_map[mobjn] += [list(d)[i]]
+            else:
+                all_pars = []
+                for mobjn in measurement_objects:
+                    # Find all sweep param names that contain the mobj name
+                    pars = [k for k in list(d) if mobjn in k]
```

Comment:
It seems to me that this could introduce bugs like adding qb12 sweep points for qb1 (or even weirder bugs if someone comes up with completely different mobj names). Shouldn't it be `if k.startswith(mobjn + '_')`?

Reply:
Ooops! I had already introduced this bug before somewhere else--good catch! :p 
But I replaced it with `k.startswith(mobj)` without the underscore because the user might now always use QE to define sweep parameter name and might have something like "qb12amplitude"

Reply:
Without the underscore, it would still add qb12 sweep points for qb1.

Reply:
Indeed ...
I wonder if you approve the solution I came up with now (after spending 2 hours ... ). In the end I don't know how to make it completely general, so now the only compromise is that it cannot find sweep parameter names that contain more than one mobjn (for example 'qb1qb9_cliffords'), which was true even in the solution using `k.startswith(mobjn + '_')`.

So now the function can handle correctly both of the cases below:

```
sweep_params_1d = ['this_qb15_amplitude1', 'this_qb11_freq1', 'this_qb1_amplitude1', 'this_qb3_freq1']
sweep_params_2d = ['this_qb15_amplitude2', 'this_qb11_freq2', 'this_qb1_amplitude2', 'this_qb3_freq2']
mobjns = ['qb3', 'qb1', 'qb11', 'qb15']
swpts.get_meas_obj_sweep_points_map(mobjns) gives
{'qb3': ['this_qb3_freq1', 'this_qb3_freq2'],
 'qb1': ['this_qb1_amplitude1', 'this_qb1_amplitude2'],
 'qb11': ['this_qb11_freq1', 'this_qb11_freq2'],
 'qb15': ['this_qb15_amplitude1', 'this_qb15_amplitude2']}
```


```
sweep_params_1d = ['this_twpaA_amplitude1', 'this_twpa_freq1', 'thistwpaB_freq1']
sweep_params_2d = ['this_twpaA_amplitude2', 'this_twpa_freq2', 'thistwpaB_freq2']
mobjns = ['twpa', 'twpaA', 'twpaB']
swpts.get_meas_obj_sweep_points_map(mobjns) gives
{'twpa': ['this_twpa_freq1', 'this_twpa_freq2'],
 'twpaA': ['this_twpaA_amplitude1', 'this_twpaA_amplitude2'],
 'twpaB': ['thistwpaB_freq1', 'thistwpaB_freq2']}
```